torch/csrc/distributed/c10d/intra_node_comm.cu

```diff
@@ -636,25 +638,33 @@ at::Tensor IntraNodeComm::hybridCubeMeshAllReduce(
 
 AllReduceAlgo IntraNodeComm::selectAllReduceAlgo(const at::Tensor& input) {
   // Only support bf16 for now
-  if (input.dtype() != at::kBFloat16 ||
-      static_cast<size_t>(input.numel() * input.element_size()) >
-          kMaxIntraNodeSize) {
+  if (input.dtype() != at::kBFloat16) {
     return AllReduceAlgo::NONE;
```

Comment:
wondering for the un-supported cases, we should fallback to nccl allreduce?

Reply:
Right. Currently, `IntraNodeComm` is wrapped by `ProcessGroupNCCL`. `ProcessGroupNCCL::all_reduce` performs algo selection and uses the returned value to determine whether to use intra-node all-reduce or normal nccl all-reduce.

I'm gradually refactoring the code to reverse the relationship. We'll have p2p process group that wraps around a nccl pg as fallback. This way the integration will not be intrusive.